OpenRA.Game/Network/SyncReport.cs

```diff
@@ -104,7 +104,8 @@ void GenerateSyncReport(Report report, IEnumerable<OrderManager.ClientOrder> ord
 
 		internal void DumpSyncReport(int frame)
 		{
-			var reportName = "syncreport-" + DateTime.UtcNow.ToString("yyyy-MM-ddTHHmmssZ", CultureInfo.InvariantCulture) + "-" + orderManager.LocalClient.Index + ".log";
+			var reportIndex = orderManager.LocalClient == null ? "" : orderManager.LocalClient.Index.ToString();
+			var reportName = "syncreport-" + DateTime.UtcNow.ToString("yyyy-MM-ddTHHmmssZ", CultureInfo.InvariantCulture) + "-" + reportIndex + ".log";
```

Comment:
```suggestion
			var reportName = "syncreport-" + DateTime.UtcNow.ToString("yyyy-MM-ddTHHmmssZ", CultureInfo.InvariantCulture) + $"-{orderManager.LocalClient?.Index}.log";
```

Reply:
Why not interpolate the entire string?

Reply:
can't have strings in strings

Reply:
mb, you actually can. I assumed c# wasn't smart enough. 
So then yes, the whole things should be interpolated

Reply:
`$"syncreport-{DateTime.UtcNow.ToString("yyyy-MM-ddTHHmmssZ", CultureInfo.InvariantCulture)}-{orderManager.LocalClient?.Index}.log"` works on dotnetfiddle.

Reply:
So something like
```suggestion
			var timestamp = DateTime.UtcNow.ToString("yyyy-MM-ddTHHmmssZ", CultureInfo.InvariantCulture);
			var reportName = $"syncreport-{timestamp}-{orderManager.LocalClient?.Index}.log";
```

Reply:
> So something like

Changed to that, feels good.